docs/yaml/functions/response_file.yaml

```diff
@@ -0,0 +1,75 @@
+name: response_file
+returns: list[str] | str
+description: |
+  Write argument list to a response file, if the resulting
+  length of the command line is longer than a maximum value.
+  A response file is a mechanism allowing a program to receive
+  more arguments than what would be allowed by the command line. See
+  [msvc compiler documentation](https://learn.microsoft.com/en-us/cpp/build/reference/at-specify-a-compiler-response-file?view=msvc-170)
+  as an example.
+
+  Returns either the list of arguments or the path to the response
+  file.
+
+  This function is designed to be used with `custom_target`,
+  `run_target`, `run_command`, `test`, or any other function
+  running a custom executable or script.
+
+varargs:
+  name: arguments
+  type: str | file | list
```

Comment:
What about build targets?

Reply:
Unfortunately from an implementation POV, this is a real mess. Meson does not have a proper common function to process command arguments in a consistent way.

You would need parts of `build.CommandBase.flatten_command()` and `backend.eval_custom_target_command()`.

Doing a proper refactoring is probably out of scope of this PR, probably good enough to do in a follow-up PR.